### PR TITLE
Update elasticsearch.rst

### PR DESCRIPTION
**Pull request by justinegeffen**
Emphasized message degradation point and when to use Elasticsearch. Updated relevant FAQ entry with the same content.

### CODE REVIEW

**Review by wiersgallak — APPROVED** [review]
Thank you!

**Review by amyblais — APPROVED** [review]

**Review by sadohert — APPROVED** [review]
Looks good.  Thanks @justinegeffen!

File: source/deployment/elasticsearch.rst [hunk 1]
```diff
@@ -3,7 +3,9 @@ Elasticsearch (E20)
 
 *Available in Enterprise Edition E20.*
 
-`Elasticsearch <https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html>`__ is a distributed, RESTful search engine supporting highly efficient database searches in a `cluster environment <https://docs.mattermost.com/deployment/cluster.html>`__.
+The Elasticsearch module in Mattermost E20 is recommended for enterprise-scale deployments to optimize search performance and avoid performance degradation and timeouts.
```
Comment (wiersgallak) [inline source/deployment/elasticsearch.rst:6]:
```suggestion
Elasticsearch provides enterprise-scale deployments optimized search performance and prevents performance degradation and timeouts. 
```

File: source/deployment/elasticsearch.rst [hunk 2]
```diff
@@ -3,7 +3,7 @@ Elasticsearch (E20)
 
 *Available in Enterprise Edition E20.*
 
-The Elasticsearch module in Mattermost E20 is recommended for enterprise-scale deployments to optimize search performance and avoid performance degradation and timeouts.
+Elasticsearch provides enterprise-scale deployments optimized search performance and prevents performance degradation and timeouts.
```
Comment (amyblais) [inline source/deployment/elasticsearch.rst]:
```suggestion
Elasticsearch provides enterprise-scale deployments for optimized search performance and prevents performance degradation and timeouts.
```
?

Comment (justinegeffen) [reply source/deployment/elasticsearch.rst]: I just saw that and added "with". I think that works?

Comment (amyblais) [reply source/deployment/elasticsearch.rst]: Yes, thanks

**Pull request merged**